diff --git a/src/embed.astro b/src/embed.astro
index ae2c1805c256081f8e95afb6d828ac07a4395062..e7854205e3ad6d614a0ecf7f08ada2d6eb9ecc26 100644
--- a/src/embed.astro
+++ b/src/embed.astro
@@ -37,12 +37,12 @@ const { embed, postUrl, compact } = Astro.props;
 			<External embed={media} compact={compact} />
 		))
 		.when(AppBskyEmbedImages.isView, (media) => (
-			<a href={postUrl}>
+			<a href={postUrl} target="_blank" rel="noopener noreferrer">
 				<ImageGrid images={media.images} />
 			</a>
 		))
 		.when(AppBskyEmbedVideo.isView, (media) => (
-			<a href={postUrl}>
+			<a href={postUrl} target="_blank" rel="noopener noreferrer">
 				<VideoThumbnail
 					thumbnail={media.thumbnail}
 					aspectRatio={media.aspectRatio}
diff --git a/src/post.astro b/src/post.astro
index dd94c79f9023aa72ae5095da262ac6e6a91a2bab..e3e9cf3113b4b3602505d296b87246b0a180e818 100644
--- a/src/post.astro
+++ b/src/post.astro
@@ -47,10 +47,10 @@ const formatter = new Intl.DateTimeFormat('en-US', {
 		<div class="post-header">
 			<Avatar user={author} link={true} />
 			<div class="user-info">
-				<a href={authorUrl} class="display-name">{author?.displayName}</a>
-				<a href={authorUrl} class="username">@{author?.handle}</a>
+				<a href={authorUrl} class="display-name" target="_blank" rel="noopener noreferrer">{author?.displayName}</a>
+				<a href={authorUrl} class="username" target="_blank" rel="noopener noreferrer">@{author?.handle}</a>
 			</div>
-			<a href={postUrl} class="logo-link">
+			<a href={postUrl} class="logo-link" target="_blank" rel="noopener noreferrer">
 				<img
 					src="data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20fill='none'%20viewBox='0%200%20320%20286'%3e%3cpath%20fill='rgb(10,122,255)'%20d='M69.364%2019.146c36.687%2027.806%2076.147%2084.186%2090.636%20114.439%2014.489-30.253%2053.948-86.633%2090.636-114.439C277.107-.917%20320-16.44%20320%2032.957c0%209.865-5.603%2082.875-8.889%2094.729-11.423%2041.208-53.045%2051.719-90.071%2045.357%2064.719%2011.12%2081.182%2047.953%2045.627%2084.785-80%2082.874-106.667-44.333-106.667-44.333s-26.667%20127.207-106.667%2044.333c-35.555-36.832-19.092-73.665%2045.627-84.785-37.026%206.362-78.648-4.149-90.071-45.357C5.603%20115.832%200%2042.822%200%2032.957%200-16.44%2042.893-.917%2069.364%2019.147Z'/%3e%3c/svg%3e"
 					class="bluesky-logo"
@@ -62,7 +62,7 @@ const formatter = new Intl.DateTimeFormat('en-US', {
 		<p class="post-text" set:html={body} />
 
 		{embed && <Embed embed={embed} postUrl={postUrl} />}
-		<a href={postUrl} class="timestamp">
+		<a href={postUrl} class="timestamp" target="_blank" rel="noopener noreferrer">
 			{formatter.format(new Date(record.createdAt ?? ''))}
 		</a>
 	</div>
diff --git a/src/quote-embed.astro b/src/quote-embed.astro
index ee570e22eb88b98e3a86ea2b4289849604077bdd..bb389e21367e39f251a29c5b58c82027663dd8c8 100644
--- a/src/quote-embed.astro
+++ b/src/quote-embed.astro
@@ -15,7 +15,7 @@ const { embed } = Astro.props;
 	AppBskyEmbedRecord.isViewRecord(embed.record) &&
 	AppBskyFeedPost.isRecord(embed.record.value) ? (
 		<div class="post-container">
-			<a href={atUriToPostUri(embed.record.uri)} class="post-link">
+			<a href={atUriToPostUri(embed.record.uri)} class="post-link" target="_blank" rel="noopener noreferrer">
 				<div class="user-info">
 					<Avatar user={embed.record.author} size="small" />
 					<p class="user-text">
@@ -24,7 +24,7 @@ const { embed } = Astro.props;
 					</p>
 				</div>
 			</a>
-			<a href={atUriToPostUri(embed.record.uri)} class="post-link">
+			<a href={atUriToPostUri(embed.record.uri)} class="post-link" target="_blank" rel="noopener noreferrer">
 				<p class="content">{embed.record.value.text}</p>
 			</a>
 			<Embed
