diff --git a/dist/cjs/embed.cjs b/dist/cjs/embed.cjs
index fc297ea59a76ca0e8017ca6f26f6cb8a1a955fd3..9e393e1de3366d7ff17748617bc4f4917c00ecda 100644
--- a/dist/cjs/embed.cjs
+++ b/dist/cjs/embed.cjs
@@ -60,7 +60,7 @@ const buildImageNode = ({ target, resolvedUrl, imageWidth, imageHeight }) => {
 	return {
 		type: "image",
 		url,
-		alt: target.page,
+		alt: "",
 		title: void 0,
 		data: hProperties.width || hProperties.height ? { hProperties } : void 0
 	};
diff --git a/dist/esm/embed.mjs b/dist/esm/embed.mjs
index dcf888cca71120674905c71e79a153620663f63c..d2da93fdfbc8032acd98d0652b0f36fe52d17881 100644
--- a/dist/esm/embed.mjs
+++ b/dist/esm/embed.mjs
@@ -58,7 +58,7 @@ const buildImageNode = ({ target, resolvedUrl, imageWidth, imageHeight }) => {
 	return {
 		type: "image",
 		url,
-		alt: target.page,
+		alt: "",
 		title: void 0,
 		data: hProperties.width || hProperties.height ? { hProperties } : void 0
 	};
